nix/pkgs.nix

```diff
@@ -1,11 +1,14 @@
 # our packages overlay
-pkgs: _: with pkgs; {
-  smashHaskellPackages = import ./haskell.nix {
-    inherit config
-      lib
-      stdenv
-      haskell-nix
-      buildPackages
-      ;
-  };
+pkgs: _: with pkgs;
+  let
+    compiler = config.haskellNix.compiler or "ghc865";
+  in {
+    smashHaskellPackages = callPackage ./haskell.nix {
+      inherit compiler;
+    };
+
+  # Grab the executable component of our package.
+  inherit (smashHaskellPackages.smash.components.exes)
```

Comment:
expose as smash-exe to make it less confusing for users to figure out how to build.